cli/tauri-bundler/src/bundle/osx_bundle.rs

```diff
@@ -87,9 +89,215 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
     create_bootstrapper(&bundle_directory, settings)
       .with_context(|| "Failed to create OSX bootstrapper")?;
   }
+
+  if let Some(identity) = settings.osx_signing_identity() {
+    sign(app_bundle_path.clone(), identity, &settings)?;
+    match notarize_auth_args() {
+      Ok(args) => {
+        notarize(app_bundle_path.clone(), args, settings)?;
+      }
+      Err(e) => {
+        common::print_info(format!("skipping app notarize, {}", e.to_string()).as_str())?;
+      }
+    }
+  }
+
   Ok(vec![app_bundle_path])
 }
 
+fn sign(app_bundle_path: PathBuf, identity: &str, settings: &Settings) -> crate::Result<()> {
+  common::print_info(format!(r#"signing app with identity "{}""#, identity).as_str())?;
+  let mut args = vec!["--deep", "--force", "-s", identity, "--options", "runtime"];
```

Comment:
Do we need to sign depth-first? `electron-osx-sign` recently began doing this:

https://github.com/electron/electron-osx-sign/blob/9bc2eedae5ece7a15576e1d564ed0ddb67a9e110/sign.js#L201-L210

Reply:
yeah i think we do, I wanted to do it but I can

Reply:
Right. I can test it for you. I'm actually not sure whether that depth-first thing was specific to Electron tbh 🤔  but I can try it out

Reply:
probably not :D I know something about Apple :joy: 

thanks!